src/app/shared/services/media-state.service.ts

```diff
@@ -0,0 +1,30 @@
+import { lastValueFrom, Observable } from 'rxjs'
+import { Injectable } from '@angular/core'
+import { MediaService } from './media.service'
+import { Media } from './media'
+import { Logger } from '../../core/abstractions/logger.service'
+
+@Injectable()
+export class MediaStateService {
+  private readonly logger: Logger
+
+  constructor(
+    private readonly mediaService: MediaService,
+    logger: Logger
+  ) {
+    this.logger = logger.tag('MediaStateService')
+  }
+
+  public async getMedia(id: string): Promise<Media> {
```

Comment:
The purpose of state services are to hold the (cached) state of entity/subject that are created for and notify users of the state, when it changes through observables. This class does not contain any state handling, so either it should be removed, and the `mediaService` should be used directly in the mini shelf, or it should be changed to a state service acting as the other state services.